### PR TITLE
Delayed job - add exclude queue name and support for ActiveRecord 2

### PR DESCRIPTION
**Pull request by jayvee**
1) New options called exclude_queue_name which gets all data for DJ except the specified queue
2) Added support for ActiveRecord 2

### CODE REVIEW

File: delayed_job/monitor_delayed_jobs.rb [hunk 1]
```diff
@@ -67,22 +70,53 @@ def build_report
...
       # Jobs that have permanently failed
-      query_hash[:failed]    = DelayedJob.find(:all, :conditions => 'failed_at IS NOT NULL')
+      conditions = ["failed_at IS NOT NULL #{queue_filter_sql}"]
+      conditions << queue_filter_params if queue_filter_params
+      query_hash[:failed]    = DelayedJob.find(:all, :conditions => conditions)
       # The oldest job that hasn't yet been run, in minutes
-      query_hash[:oldest]    = DelayedJob.find(:all, :conditions => [ 'run_at <= ? AND locked_at IS NULL AND attempts = 0', Time.now.utc ], :order => :run_at)
+      conditions = ["run_at <= ? AND locked_at IS NULL AND attempts = 0 #{queue_filter_sql}", Time.now.utc]
+      conditions << queue_filter_params if queue_filter_params
+      query_hash[:oldest]    = DelayedJob.find(:all, :conditions => conditions, :order => :run_at)
```
Comment (ejoubaud) [inline delayed_job/monitor_delayed_jobs.rb]: All those statements look very much alike. They could be DRYed up with something like:

```ruby
{
  :running => { :sql => "locked_at IS NOT NULL AND failed_at IS NULL"},
  :waiting => { :sql => "run_at <= ? AND locked_at IS NULL AND attempts = 0", :args => [ Time.now.utc ] },
  ...
}.each do |name, critera|
  sql = criteria[:sql] + queue_filter_sql
  args = criteria.fetch(:args, [])
  args << queue_filter_params if queue_filter_params
  query_hash[name] = DelayedJob.find(:all, :conditions => args.unshift(sql))
end
```

File: delayed_job/monitor_delayed_jobs.rb [hunk 2]
```diff
@@ -67,22 +70,53 @@ def build_report
           query_hash[key] = query_hash[key].where('queue = ?', option(:queue_name))
         end
       end
+      if option(:exclude_queue_name)
```
Comment (ejoubaud) [inline delayed_job/monitor_delayed_jobs.rb]: Nitpick: Considering they both update the same hash key, we could save a line with `elsif`

File: delayed_job/monitor_delayed_jobs.rb [hunk 3]
```diff
@@ -67,22 +70,53 @@ def build_report
...
+      end
     else
       # ActiveRecord 2.x compatible
+      queue_filter_sql = ""
+      queue_filter_params = nil
+      if option(:queue_name)
+        queue_filter_sql = " and queue = ?"
+        queue_filter_params = option(:queue_name)
+      end
+      if option(:exclude_queue_name)
```
Comment (ejoubaud) [inline delayed_job/monitor_delayed_jobs.rb]: Same here, `elsif` could save a line.

**Comment by ejoubaud** [comment]
:+1:

**Comment by warrenseen** [comment]
:+1: Looks good to me.

**Comment by johnsyweb** [comment]
:+1: LGTM, too!

**Pull request merged**